contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,1375 @@
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventStarted {
+                start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+
+        // Edge case current time is event start
+        env.block.time = Timestamp::from_seconds(EVENT_START_SECONDS);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventStarted {
+                start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+    }
+
+    #[test]
+    fn event_info_update_after_event_terminated_error() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        let msg = ExecuteMsg::UpdateEventInfo {
+            start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+            // Add 300 seconds to prevent end time to be already passed
+            end_time: Timestamp::from_seconds(EVENT_END_SECONDS + 300),
+        };
+
+        // Fake current time to event ended
+        env.block.time = Timestamp::from_seconds(EVENT_END_SECONDS + 100);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventTerminated {
+                end_time: Timestamp::from_seconds(EVENT_END_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+
+        // Edge case current time is event end
+        env.block.time = Timestamp::from_seconds(EVENT_END_SECONDS);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventTerminated {
+                end_time: Timestamp::from_seconds(EVENT_END_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+    }
+
+    #[test]
+    fn event_info_update_properly() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        let msg = ExecuteMsg::UpdateEventInfo {
+            start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+            end_time: Timestamp::from_seconds(EVENT_END_SECONDS + 300),
+        };
+
+        // Current time is before event started
+        env.block.time = Timestamp::from_seconds(EVENT_START_SECONDS - 100);
+
+        execute(deps.as_mut(), env, info, msg).unwrap();
+    }
+
+    #[test]
+    fn event_info_start_time_equal_end_time_error() {
```

Comment:
Similar as above, `event_info_start_time_equal_end_time_error`, `event_info_start_time_after_end_time_error` can be merged into `update_event_info_with_invalid_msg_error` with one case.

Reply:
As above also these tests were also a single one and then they were split